azure_data_cosmos_engine/src/query/producer/read_many.rs

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use std::collections::{HashMap, VecDeque};
+
+use crate::{
+    query::{node::PipelineNodeResult, query_result::QueryResultShape, DataRequest, QueryResult},
+    ErrorKind,
+};
+
+use super::{create_query_chunk_states, state::QueryChunkState};
+
+pub struct ReadManyStrategy {
+    pub query_chunks: Vec<HashMap<String, Vec<(usize, String, String)>>>,
+    pub current_query_chunk_index: usize,
+    pub query_chunk_states: Vec<QueryChunkState>,
```

Comment:
Could this be merged into the `query_chunks` property and just store a single list of `QueryChunkState` which includes both the relevant info AND the pagination state?